src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/MetricFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+import com.amazon.opendistroforelasticsearch.sql.query.join.BackOffRetryStrategy;
+
+public class MetricFactory {
+
+    public static Metric createMetric(MetricName name) {
```

Comment:
Should I be able to create more than one `Metric` instance for each `MetricName` ?

Reply:
No, Metric is singleton for SQL plugin.

Reply:
`Metrics` is, but `Metric` and its subclasses are not.

Reply:
Oh, I got what you mean. Metric is not singleton, but we only hold one metric per MetricName.

Reply:
for now yes, but you don't restrict it. I think we should not allow registering metrics with the same name, moreover, not even creating if there is already a registered metric.

Reply:
We could override the metric, just one metric for one name is OK.